src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -1999,7 +2062,60 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
     out.source.edge_target_set <+ new_edge_target;
 
 
+    // == Multi Connect
+    trace out.edge_creation_mode;
+
+    port_input_mouse_up_multi  <- inputs.hover_node_input.sample(&inputs.edge_multi_connect_up).unwrap();
+    port_output_mouse_up_multi <- inputs.hover_node_output.sample(&inputs.edge_multi_connect_up).unwrap();
+
+    trace port_input_mouse_up_multi;
+    trace port_output_mouse_up_multi;
+
+    new_edge_source_multi <= port_output_mouse_up_multi.gate(&out.edge_creation_mode).map(f!((target) {
+        model.set_detached_edge_source(target)
+    }));
+    new_edge_target_multi <= port_input_mouse_up_multi.gate(&out.edge_creation_mode).map(f!((target) {
+        model.set_detached_edge_target(target)
+    }));
+
+    re_create_edge_from_source <- new_edge_source_multi.gate(&out.edge_multi_connect)._0();
+    re_create_edge_from_target <- new_edge_target_multi.gate(&out.edge_multi_connect)._0();
```

Comment:
prefix "re_" doesnt tell much. If you want short names, you can abbreviate "create" with "mk", which is very common (thanks unixes!). The same applies to `tgt` and `src`. We are alwyas suepr ok with such helper variable names (until they become public api).